diff --git a/rules/rspamd.lua b/rules/rspamd.lua
index a193eb495..40b6af704 100644
--- a/rules/rspamd.lua
+++ b/rules/rspamd.lua
@@ -21,7 +21,7 @@ require "global_functions" ()
 config['regexp'] = {}
 rspamd_maps = {} -- Global maps

-local local_conf = rspamd_paths['CONFDIR']
+local local_conf = rspamd_paths['LOCAL_CONFDIR']
 local local_rules = rspamd_paths['RULESDIR']
 local rspamd_util = require "rspamd_util"

diff --git a/src/lua/lua_common.c b/src/lua/lua_common.c
index 323957086..1e4538b1c 100644
--- a/src/lua/lua_common.c
+++ b/src/lua/lua_common.c
@@ -576,6 +576,7 @@ rspamd_lua_set_globals (struct rspamd_config *cfg, lua_State *L,
 	lua_getglobal (L, "rspamd_paths");
 	if (lua_isnil (L, -1)) {
 		const gchar *confdir = RSPAMD_CONFDIR, *rundir = RSPAMD_RUNDIR,
+				*local_confdir = RSPAMD_LOCAL_CONFDIR,
 				*dbdir = RSPAMD_DBDIR, *logdir = RSPAMD_LOGDIR,
 				*wwwdir = RSPAMD_WWWDIR, *pluginsdir = RSPAMD_PLUGINSDIR,
 				*rulesdir = RSPAMD_RULESDIR, *lualibdir = RSPAMD_LUALIBDIR,
@@ -623,6 +624,11 @@ rspamd_lua_set_globals (struct rspamd_config *cfg, lua_State *L,
 			confdir = t;
 		}

+		t = getenv ("LOCAL_CONFDIR");
+		if (t) {
+			local_confdir = t;
+		}
+

 		if (vars) {
 			t = g_hash_table_lookup (vars, "PLUGINSDIR");
@@ -655,6 +661,11 @@ rspamd_lua_set_globals (struct rspamd_config *cfg, lua_State *L,
 				confdir = t;
 			}

+			t = g_hash_table_lookup (vars, "LOCAL_CONFDIR");
+			if (t) {
+				local_confdir = t;
+			}
+
 			t = g_hash_table_lookup (vars, "DBDIR");
 			if (t) {
 				dbdir = t;
@@ -669,6 +680,7 @@ rspamd_lua_set_globals (struct rspamd_config *cfg, lua_State *L,
 		lua_createtable (L, 0, 9);

 		rspamd_lua_table_set (L, RSPAMD_CONFDIR_INDEX, confdir);
+		rspamd_lua_table_set (L, RSPAMD_LOCAL_CONFDIR_INDEX, local_confdir);
 		rspamd_lua_table_set (L, RSPAMD_RUNDIR_INDEX, rundir);
 		rspamd_lua_table_set (L, RSPAMD_DBDIR_INDEX, dbdir);
 		rspamd_lua_table_set (L, RSPAMD_LOGDIR_INDEX, logdir);
diff --git a/src/lua/lua_common.h b/src/lua/lua_common.h
index 838e0fe7a..5810184b7 100644
--- a/src/lua/lua_common.h
+++ b/src/lua/lua_common.h
@@ -410,6 +410,7 @@ gboolean rspamd_lua_require_function (lua_State *L, const gchar *modname,

 /* Paths defs */
 #define RSPAMD_CONFDIR_INDEX "CONFDIR"
+#define RSPAMD_LOCAL_CONFDIR_INDEX "LOCAL_CONFDIR"
 #define RSPAMD_RUNDIR_INDEX "RUNDIR"
 #define RSPAMD_DBDIR_INDEX "DBDIR"
 #define RSPAMD_LOGDIR_INDEX "LOGDIR"
diff --git a/src/lua/lua_config.c b/src/lua/lua_config.c
index 2093cbe01..d5d80914b 100644
--- a/src/lua/lua_config.c
+++ b/src/lua/lua_config.c
@@ -3339,6 +3339,7 @@ lua_config_load_ucl (lua_State *L)

 		if (lua_istable (L, -1)) {
 			LUA_TABLE_TO_HASH(paths, RSPAMD_CONFDIR_INDEX);
+			LUA_TABLE_TO_HASH(paths, RSPAMD_LOCAL_CONFDIR_INDEX);
 			LUA_TABLE_TO_HASH(paths, RSPAMD_RUNDIR_INDEX);
 			LUA_TABLE_TO_HASH(paths, RSPAMD_DBDIR_INDEX);
 			LUA_TABLE_TO_HASH(paths, RSPAMD_LOGDIR_INDEX);
